core/src/main/java/org/ldaptive/ssl/X509CertificatesCredentialReader.java

```diff
@@ -1,24 +1,46 @@
 /* See LICENSE for licensing and NOTICE for copyright. */
 package org.ldaptive.ssl;
 
+
 import java.io.IOException;
 import java.io.InputStream;
 import java.security.GeneralSecurityException;
 import java.security.cert.CertificateFactory;
 import java.security.cert.X509Certificate;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 
 /**
- * Loads X.509 certificate credentials from a classpath, filepath, or stream resource. Supported certificate formats
- * include: PEM, DER, and PKCS7.
+ * Loads X.509 certificate credentials from a classpath, filepath, or stream resource.
+ * When working with filepath, multiple files may be separated using a comma (i.e. {@code cert1.pem,cert2.crt}).
+ * Supported certificate formats include: PEM, DER, and PKCS7.
  *
  * @author  Middleware Services
  */
 public class X509CertificatesCredentialReader extends AbstractCredentialReader<X509Certificate[]>
 {
 
-
+  @Override
+  public X509Certificate[] read(final String path, final String... params)
+      throws IOException, GeneralSecurityException
+  {
+    final String[] paths = path.split(",");
```

Comment:
I think I prefer that the `path` argument be an array, then the path delimiter is an implementation detail for the invoker.

Reply:
Sure. Need some help with this. 

If I switch to:

```
public T read(final String[] paths, final String... params) {}
```

That means I'd have to change the parent interface, which means I'd have to remove this method and fix the parent `AbstractCredentialReader` as well, and with that I am not how how to handle the return type. Am I on the right path?

Reply:
I wasn't going to update the interface or abstract class for this. The ability to parse multiple paths is unique to this one class currently, so I was expecting to add
```
public X509Certificate[] read(final String[] paths, final String... params)
```
directly to `X509CertificatesCredentialReader`

If other implementations need this in the future we can revisit. Thoughts?

Reply:
If you're leveraging the interface or some other component, I'll need to rethink.

Reply:
And I can see if you want to use this with `X509CredentialConfig` (which you probably do). This isn't going work. Nevermind.

I think the only thing to change here is maybe use the platform path separator rather than defaulting to comma.

Reply:
Thanks for the notes. Sure, will make the change.

Reply:
Sorry to bother again; more questions :)

According to the docs, the path separator character,

> It is a semicolon (;) on Windows and colon (:) on Mac and Unix-based operating systems.

The issue here is that paths as `Resource`s require a classifier/prefix. As in, `file:/path/to.xyz.pem`. This messes up the split function on Mac and Unix. Not sure how to solve this without manually parsing stuff. Would be best to fall back onto comma or space or is there a better way I am missing here?

Reply:
I would probably use `space`. My second choice would be `comma`. But it's hard to say which would be more problematic. Does spring have a convention?

Reply:
Judging by the function in Spring's StringUtils (i.e `commaDelimitedListToStringArray`) and loads of other stuff in that class that deal with CSVs, I think Spring would go for comma most likely. I'd think space would be more problematic on windows, since you'd have paths like "C:\Program Files", and generally it's more accepted on windows to have spaces in paths.

Reply:
+1 for comma.